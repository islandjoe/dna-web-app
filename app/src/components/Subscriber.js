import React, {Component} from 'react'
import {subscribers} from '../data'

class Subscriber extends Component {

  constructor(props) {
    super(props)

    this.state = {
      userid: props.userid,
      name: '',
      address: '',
      mobileNumber: '',
      subscriptions: []
    }
  }

  componentDidMount() {
    const subscriber = subscribers.find((s)=> s.userid === this.state.userid)

    if (subscriber) {
      this.setState({ ...subscriber })
    }
  }

  render() {
    return (
      <h2>{ this.state.name }</h2>
    )
  }
}

export default Subscriber
