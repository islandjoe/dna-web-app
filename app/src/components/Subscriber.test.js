import React from 'react'
import {shallow} from  'enzyme'
import Subscriber from './Subscriber.js'

it('should render correctly', ()=> {
  const component = shallow(<Subscriber/>)
  expect(component).toMatchSnapshot()
})