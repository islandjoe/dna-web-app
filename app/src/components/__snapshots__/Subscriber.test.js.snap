// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<styled.h1>
  Arthur Kho Caayon
</styled.h1>
`;
